### PR TITLE
[FIX] mass_mailing: fix the merge mailing list

### PR DESCRIPTION
**Pull request by dht-odoo**
PURPOSE:

To resolve the error occurring while merging the mail list and change 
name of button to "MERGE" from "Merge Selected Mailing Lists".

SPECIFICATION:

Current:
Error: 'int' object is not subscriptable
Button name: "MERGE" from "Merge Selected Mailing Lists".

To be:
To resolve this error and change button name to "MERGE"

LINKS:

Task Id: 2471692
PR: #67213

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mass_mailing/wizard/mailing_list_merge.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def default_get(self, fields):
         res = super(MassMailingListMerge, self).default_get(fields)
         src_list_ids = self.env.context.get('active_ids')
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/wizard/mailing_list_merge.py:23]: While we are at it, we could check that active_model is effectively mailing.list .

File: addons/mass_mailing/models/mailing_list.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def _compute_contact_nbr(self):
             ''', (tuple(self.ids), ))
             data = dict(self.env.cr.fetchall())
             for mailing_list in self:
-                mailing_list.contact_nbr = data.get(mailing_list.id, 0)
+                mailing_list.contact_nbr = data.get(mailing_list._origin.id, 0)
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/models/mailing_list.py:45]: Isn't it linked to the 'if self.ids else 0' global structure of the compute ? Maybe we should use self.ids or [self._origin.id] in the compute.

File: addons/mass_mailing/wizard/mailing_list_merge.py [hunk 3]
```diff
@@ -20,9 +21,14 @@ class MassMailingListMerge(models.TransientModel):
     @api.model
     def default_get(self, fields):
         res = super(MassMailingListMerge, self).default_get(fields)
+
+        active_model = self.env.context.get('active_model')
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/wizard/mailing_list_merge.py:25]: Could be inlined directly in the herebelow condition

File: addons/mass_mailing/wizard/mailing_list_merge.py [hunk 4]
```diff
@@ -20,9 +21,14 @@ class MassMailingListMerge(models.TransientModel):
...
+        if active_model != 'mailing.list':
+            raise UserError(_('You can only apply this action from Mailing Lists.'))
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/wizard/mailing_list_merge.py:27]: Don't forget to export the new string

**Review by tde-banana-odoo — COMMENTED** [review]
Post-question validation !

File: addons/mass_mailing/models/mailing_list.py (hunk 2 above)
Comment (dja-odoo) [reply addons/mass_mailing/models/mailing_list.py:45]: Compute method here follows that structure already. We do the computation only if there are `self.ids`, and yet we get `NewId` in self.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**